### DISCUSSION TITLE
Read only table

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6832] [General]
Hi,

Is it possible to set table as read only, so that user can only read data, not insert/modify/delete?

Thank you,
Elena.
![Schermata 2023-07-13 alle 16 27 10](https://github.com/serenity-is/Serenity/assets/14870650/f43cc599-074d-4645-bc22-fd8c91f612a8)

**Comment by edwardch** [comment]
Check out https://demo.serenity.is/BasicSamples/RemovingAddButton

To remove the edit link, go to your ErrorColumns.cs file and remove the `[EditLink]` attributes from the Errore column. You will also want to remove this from the ID column.

Also make sure you put the right permissions on the Row.

If there is a separate system that inserts these records to the database, you can actually remove the Save and Delete handlers too (you would need to remember to remove them from the Endpoint as well).

hth
